### DISCUSSION TITLE
How to decode save files

### DISCUSSION THREAD

**Discussion by EBGU** [kwsch/PKHeX#3558] [Q&A] [answered]
First of all, PKHex is really an interesting tool, and I love it!
I just wonder how your guys decode the save file of pokemon games.
I have tried to make some changes in the game and compare files with a hex editor, but it didn't work. Therefore, is there any tutorial teaching me to understand/decode save files?
Thank you so much!

**Comment by ReignOfComputer** [comment]
https://github.com/kwsch/PKHeX/tree/master/PKHeX.Core/Saves/Encryption might help?

**Comment by kwsch** [comment, marked as answer]
Each set of games has its own format, and quirks associated with to ensure the integrity of the data. Stuff like checksums and cryptographic signatures are used to double check that the data was not externally mutated.

PKHeX has abstractions for each format, so you can poke around the implementations to see how the program modifies and exports the save data.
